config/routes.rb

```diff
@@ -3,4 +3,7 @@
   resources :users, only: [:index, :show] do
     resources :posts, only: [:index, :show]
   end
+  resources :posts, only: [:index, :show, :new, :create]
+  resources :comments, only: [:new, :create]
+  resources :likes, only: [:create]
 end
```

Comment:
[OPTIONAL] You have done an excellent job, kudos for using partials! 😎 Just a suggestion, you could add `:new` inside the nested routes:
```
 resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show, :new]
  end
```
Now that you have the `current_user` you can create the post on its behalf. I'm just saying this because when you say
`resources :posts, only: [:index, :show, :new, :create]` you could get some errors when you try to use `post_path` because there are two paths: `/users/:user_id/posts/:post_id` and `/posts/:posts_id`

Reply:
> [OPTIONAL] You have done an excellent job, kudos for using partials! 😎 Just a suggestion, you could add `:new` inside the nested routes:
> 
> ```
>  resources :users, only: [:index, :show] do
>     resources :posts, only: [:index, :show, :new]
>   end
> ```
> 
> Now that you have the `current_user` you can create the post on its behalf. I'm just saying this because when you say `resources :posts, only: [:index, :show, :new, :create]` you could get some errors when you try to use `post_path` because there are two paths: `/users/:user_id/posts/:post_id` and `/posts/:posts_id`

Thank you @smunozmo for your comment, I understand what you're pointing to, but I handled those cases in the `PostController`.

Reply:
@teckim sure! Well done! 🤓 🚀